tests/data/test_temporal_random_walker.py

```diff
@@ -81,3 +92,14 @@ def test_exp_biases_extreme(temporal_graph):
     small_times = [0.000001, 0.000002]
     biases = rw._exp_biases(small_times, t_0=0, decay=True)
     assert sum(biases) == pytest.approx(1)
+
+
+@pytest.mark.parametrize("num_edges", [1, 10, 20])
+def test_validate_times(num_edges):
+    with pytest.raises(ValueError, match=r".*edge times must be non-negative.*") as err:
+        TemporalRandomWalk(temporal_graph_negative_times(num_edges))
+
+    err_msg = err.value.args[0]
+    assert str(num_edges) in err_msg
```

Comment:
As alternative to this manual checking, we could dynamically construct the appropriate regex:

```python
trailing = r"\.\.\." if num_edges > 10 else ""
matcher = fr".*edge times must be non-negative.*{num_edges}.*{trailing}$"

with pytest.raises(ValueError, match=matcher): 
    ...
```

Reply:
Nice 👍